src/components/ItemEditorPage/LeftPanel/Header/Header.tsx

```diff
@@ -15,8 +15,13 @@ export default class Header extends React.PureComponent<Props> {
   }
 
   handleBack = () => {
-    const { onNavigate } = this.props
-    onNavigate(locations.itemEditor())
+    const { collection, isFromCollections, onNavigate } = this.props
+    // If the user came from a collection details page, go back to the same page
+    if (isFromCollections) {
+      onNavigate(locations.collectionDetail(collection!.id))
```

Comment:
can we avoid this `collection!` by adding `collection && isFromCollections` in the `if` above?